### PR TITLE
[MOB-5762] - [Manager 1] - Time based AutoSync

### PR DESCRIPTION
**Pull request by Ayyanchira**
Schedule sync happens after every determined duration. Still not perfect. Especially in situation of going to background and coming to foreground in time less than the autofetch interval. Possiblity of multiple looper posts

## 🔹 Jira Ticket(s) if any

* [MOB-XXXX](https://iterable.atlassian.net/browse/MOB-XXXX)

## ✏️ Description

> Please provide a brief description of what this pull request does.

### CODE REVIEW

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableConfig.java [hunk 1]
```diff
@@ -71,6 +71,11 @@ public class IterableConfig {
      */
     final String[] allowedProtocols;
 
+    /**
+     * The number of seconds to wait before fetching the next embedded message
+     */
+    final double embeddedMessageAutoFetchInterval;
```
Comment (roninopf) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableConfig.java:77]: Should we name this `embedded**Messaging**AutoFetchInterval`?

Comment (Ayyanchira) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableConfig.java:77]: i miss flex here 🥲 😁

**Review by roninopf — APPROVED** [review]

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableEmbeddedManager.kt [hunk 2]
```diff
@@ -1,58 +1,59 @@
...
-
-    var autoFetchDuration: Int = 0
-    var timer: Timer? = null
-
-    var actionHandler: EmbeddedMessageActionHandler? = null
-    var updateHandler: EmbeddedMessageUpdateHandler? = null
-
-    var actionHandleListeners = mutableListOf<EmbeddedMessageActionHandler>()
-    var updateHandleListeners = mutableListOf<EmbeddedMessageUpdateHandler>()
+public class IterableEmbeddedManager: IterableActivityMonitor.AppStateCallback{
```
Comment (evangreer91) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableEmbeddedManager.kt:12]: what does this do?

Comment (evangreer91) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableEmbeddedManager.kt:12]: Oh I see this extends with the AppStateCallback so we can define onSwitchToForeground and onSwitchToBackground overrides

**Pull request merged**